src/log_dumper.rs

```diff
@@ -0,0 +1,35 @@
+use crate::file_pipe_log::ReplayMachine;
+use crate::log_batch::{LogItem, LogItemBatch};
+use crate::{FileId, LogQueue};
+
+use std::collections::HashMap;
+
+pub struct LogDumer {
+    // raft_groups_id to LogItem vec
+    pub(crate) logs: HashMap<u64, Vec<LogItem>>,
```

Comment:
Use `LogItemReader` instead:
https://github.com/tikv/raft-engine/blob/e32e3e78604b7115621f543dea1231132abd8778/src/file_pipe_log/mod.rs#L48
You should wrap it into a `std::iter::Iter` type.

Reply:
🆗